version: '3.8'

services:
  db:
    networks:
     - pg-network
    image: postgres:13
    container_name: postgres_db
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: testdb
    ports:
      - "5433:5432"
    volumes:
      - db_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 1s
      timeout: 5s
      retries: 10

  app:
    networks:
     - pg-network
    build:
      context: .
    depends_on:
      db:
        condition: service_healthy
    container_name: python_app_it2

volumes:
  db_data:
networks:
  pg-network:
